error.go

```diff
@@ -6,82 +6,72 @@
+	ErrMemorySwapLimit               = lxcError("your kernel does not support cgroup swap controller")
+	ErrMethodNotAllowed              = lxcError("the requested method is not currently supported with unprivileged containers")
+	ErrNewFailed                     = lxcError("allocating the container failed")
+	ErrNoSnapshot                    = lxcError("container has no snapshot")
+	ErrNotDefined                    = lxcError("container is not defined")
+	ErrNotFrozen                     = lxcError("container is not frozen")
+	ErrNotRunning                    = lxcError("container is not running")
+	ErrNotSupported                  = lxcError("method is not supported by this LXC version")
+	ErrRebootFailed                  = lxcError("rebooting the container failed")
+	ErrRemoveDeviceNodeFailed        = lxcError("removing device from container failed")
+	ErrRenameFailed                  = lxcError("renaming the container failed")
+	ErrRestoreFailed                 = lxcError("restore failed")
+	ErrRestoreSnapshotFailed         = lxcError("restoring the container failed")
+	ErrSaveConfigFailed              = lxcError("saving config file for the container failed")
+	ErrSettingCgroupItemFailed       = lxcError("setting cgroup item for the container failed")
+	ErrSettingConfigItemFailed       = lxcError("setting config item for the container failed")
+	ErrSettingConfigPathFailed       = lxcError("setting config file for the container failed")
+	ErrSettingKMemoryLimitFailed     = lxcError("setting kernel memory limit for the container failed")
+	ErrSettingMemoryLimitFailed      = lxcError("setting memory limit for the container failed")
+	ErrSettingMemorySwapLimitFailed  = lxcError("setting memory+swap limit for the container failed")
+	ErrSettingSoftMemoryLimitFailed  = lxcError("setting soft memory limit for the container failed")
+	ErrShutdownFailed                = lxcError("shutting down the container failed")
+	ErrSoftMemLimit                  = lxcError("your kernel does not support cgroup memory controller")
+	ErrStartFailed                   = lxcError("starting the container failed")
+	ErrStopFailed                    = lxcError("stopping the container failed")
+	ErrTemplateNotAllowed            = lxcError("unprivileged users only allowed to use \"download\" template")
+	ErrUnfreezeFailed                = lxcError("unfreezing the container failed")
+	ErrUnknownBackendStore           = lxcError("unknown backend type")
+	ErrReleaseFailed                 = lxcError("releasing the container failed")
 )
 
-// Error represents a basic error that implies the error interface.
-type Error struct {
-	Message string
-}
-
-// NewError creates a new error with the given msg argument.
-func NewError(msg string) error {
-	return &Error{
-		Message: msg,
-	}
-}
+type lxcError string
 
-func (e *Error) Error() string {
-	return e.Message
+func (e lxcError) Error() string {
```

Comment:
Hm, isn't that an API break?

Reply:
In theory this should not break API (famous last words) All the checks like `if err == ErrAllocationFailed` or switch statements etc. should work as before.